File: Python/climbing-stairs.py
class Solution:
    def climbStairs(self, n):
        ans = [1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144]
        for i in range(11, n):
            ans.append(ans[i-1] + ans[i-2])
        return ans[n-1]

File: C/climbing-stairs.c
int climbStairs(int n){
    int i;
    int dp[n + 1];
    dp[0] = 1;
    dp[1] = 1;
    for (i = 2; i < n + 1; ++i) {
        dp[i] = dp[i - 2] + dp[i - 1];
    }

    return dp[n];
}
